file-formats/dobj/type/zif_aff_dobj_v1.intf.abap

```diff
@@ -0,0 +1,61 @@
+INTERFACE zif_aff_dobj_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">Structure Definition</p>
+    "! Structure Definition
+    BEGIN OF ty_structure_definition,
+      "! <p class="shorttext">Structure Definition</p>
+      "! Parent Table
+      "! $required
+      record_no       TYPE n LENGTH 4,
+      "! <p class="shorttext">Parent Table</p>
+      "! Parent Table
+      "! $required
+      parent_table    TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Dependent table</p>
+      "! Dependent table
+      "! $required
+      dependent_table TYPE zif_aff_types_v1=>ty_object_name_30,
+    END OF ty_structure_definition,
+    "! <p class="shorttext">Structure Definitions</p>
+    "! Structure Definitions
+    ty_structure_definitions TYPE SORTED TABLE OF  ty_structure_definition WITH UNIQUE KEY record_no.
+
+
+  TYPES:
+    "! <p class="shorttext">Data destruction object</p>
+    "! $required
+    BEGIN OF ty_dobj_details,
+      "! <p class="shorttext">Destruction Program</p>
+      "! Destruction Program
+      "! $required
+      destr_program        TYPE zif_aff_types_v1=>ty_description_60,
+      "! <p class="shorttext">Application Component</p>
+      "! Application Component
+      applic_component     TYPE zif_aff_types_v1=>ty_object_name_30,
```

Comment:
What is the use case of the application component? Is it correct that it refers to an object name?

Abbreviation?

Reply:
Application component refers to the component to which class/program belongs to. This application component is of length 24. So I used the type zif_aff_types_v1=>ty_object_name_30 for the application component. Is that correct?